File: src/Domain/Entities/TestCase.cs
using Domain.Enums;

namespace Domain.Entities;

public class TestCase : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public TestCasePriority Priority { get; set; } = TestCasePriority.Medium;

    public Guid SuiteId { get; set; }
    public TestSuite Suite { get; set; } = null!;

    public ICollection<TestCaseStep> Steps { get; set; } = [];
    public ICollection<TestResult> TestResults { get; set; } = [];
}

File: apps/api/src/domain/test-case.ts
import { TestCasePriority } from "@testcraft/types";

export interface TestCase {
  id: string;
  suiteId: string;
  name: string;
  description: string | null;
  priority: TestCasePriority;
  stepCount: number;
  createdAt: Date;
  updatedAt: Date;
}
